tests/tests.bats

```diff
@@ -3,76 +3,73 @@
 RUN_SCRIPT='./bin/run.sh'
 DATA_DIR='tests/data'
 
+actual_matches_expected() {
+    local actual=$1 expected=$2
+
+    # Look at parts of the results instead of just comparing the file contents:
+    # when we upgrade the base image of the test container and the tool
+    # versions change, we won't need to go back and "fix" the expected result
+    # files.
+
+    [[ "$(jq .version "$actual")" == "$(jq .version "$expected")" ]] &&
+    [[ "$(jq .status "$actual")" == "$(jq .status "$expected")" ]] &&
+    [[ "$(jq .tests "$actual")" == "$(jq .tests "$expected")" ]] &&
+    [[ -n "$(jq '."test-environment".bash' "$actual")" ]] &&
+    [[ -n "$(jq '."test-environment".bats' "$actual")" ]]
+}
+
 @test "one passing test" {
     TEST="one_passing"
     TEST_DIR="$DATA_DIR/$TEST"
     run "$RUN_SCRIPT" "$TEST" "$TEST_DIR" "$TEST_DIR"
 
-    RESULTS=$(cat "$TEST_DIR/results.json")
-    EXPECTED_RESULTS=$(cat "$TEST_DIR/expected_results.json")
-
     [[ "$status" -eq 0 ]]
-    [[ "$RESULTS" == "$EXPECTED_RESULTS" ]]
+    actual_matches_expected "$TEST_DIR"/{,expected_}results.json
```

Comment:
Expanding the paths by hand might be easier to read than using a brace expansion.